### PR TITLE
feat(pci.storages.databases): datatr-377 - rework availabilities

### PR DESCRIPTION
**Pull request by jperchoc**
DATATR-377

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #DATATR-377
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Reflect major changes on availabilities api response in order to allow new mongodb flavors.

### CODE REVIEW

File: packages/manager/modules/pci/src/components/project/storages/databases/availability.class.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { find, get } from 'lodash';
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/components/project/storages/databases/availability.class.js:1]: **suggestion:** Use native JS capabilities like "Optional Chaining"

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/plans-list/plans-list.controller.js [hunk 2]
```diff
@@ -7,6 +7,10 @@ export default class {
     this.$translate = $translate;
   }
 
+  getSortedPlans() {
+    return this.plans?.sort((a, b) => b.compare(a));
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/components/plans-list/plans-list.controller.js:11]: **question:** Where is `compare` function defined?

File: packages/manager/modules/pci/src/components/project/storages/databases/availability.class.js (hunk 1 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/components/project/storages/databases/availability.class.js]: Changed here: https://github.com/ovh/manager/pull/10142/commits/a9dc304f4dc738deb8767082d332ee5a815b61ed

File: packages/manager/modules/pci/src/projects/project/storages/databases/components/plans-list/plans-list.controller.js (hunk 2 above)
Comment (jperchoc) [reply packages/manager/modules/pci/src/projects/project/storages/databases/components/plans-list/plans-list.controller.js:11]: It is defined in the `plan.class.js` file: https://github.com/ovh/manager/blob/a9dc304f4dc738deb8767082d332ee5a815b61ed/packages/manager/modules/pci/src/components/project/storages/databases/plan.class.js#L108

**Review by abullet33 — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by abullet33 — APPROVED** [review]

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js [hunk 3]
```diff
@@ -39,19 +39,35 @@ export default class {
       PRIVATE_NETWORK_GUIDE[this.user.ovhSubsidiary] ||
       PRIVATE_NETWORK_GUIDE.DEFAULT;
 
+    const availablePlans = [
+      ...this.availableEngines
+        .find((e) => e.name === this.database.engine)
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js:44]:
```suggestion
        .find((engine) => engine.name === this.database.engine)
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js [hunk 4]
```diff
@@ -39,19 +39,35 @@ export default class {
...
+        .versions.find((v) => v.version === this.database.version).plans,
+    ];
+    if (!availablePlans.find((p) => p.name === this.database.plan)) {
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js:47]:
```suggestion
    if (!availablePlans.find((plan) => plan.name === this.database.plan)) {
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js [hunk 5]
```diff
@@ -39,19 +39,35 @@ export default class {
...
+      const allPlans = this.engine.getAvailablePlans(
+        this.database.version,
+        this.database.region,
+      );
+      const currPlan = allPlans.find((p) => p.name === this.database.plan);
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js:52]:
```suggestion
      const currentPlan = allPlans.find((plan) => plan.name === this.database.plan);
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js [hunk 6]
```diff
@@ -39,19 +39,35 @@ export default class {
...
+      availablePlans.push(currPlan);
+    }
+
     // Retrieve data from db object
     const engine = this.database.getEngineFromList(this.availableEngines);
     const version = engine.getVersion(this.database.version);
     engine.selectedVersion = version;
-    this.originalPlan = version.getPlan(this.database.plan);
+    this.originalPlan = availablePlans.find(
+      (p) => p.name === this.database.plan,
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/backups/fork/fork.controller.js:61]:
```suggestion
      (plan) => plan.name === this.database.plan,
```

File: packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/upgrade-plan/upgrade-plan.routing.js [hunk 7]
```diff
@@ -8,8 +8,19 @@ export default /* @ngInject */ ($stateProvider) => {
...
-        plans: /* @ngInject */ (database, engine) =>
-          engine.getAvailablePlans(database.version, database.region),
+        plans: /* @ngInject */ (database, currentPlan, availableEngines) => {
+          const availablePlans = [
+            ...availableEngines
+              .find((engine) => engine.name === database.engine)
+              .versions.find((version) => version.version === database.version)
+              .plans,
+          ];
+          if (!availablePlans.find((p) => p.name === currentPlan.name)) {
```
Comment (anooparveti) [inline packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/upgrade-plan/upgrade-plan.routing.js:18]:
```suggestion
          if (!availablePlans.find((plan) => plan.name === currentPlan.name)) {
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Pull request merged**